org.metaborg.spoofax.core/src/main/java/org/metaborg/spoofax/core/stratego/StrategoCommon.java

```diff
@@ -112,6 +155,35 @@
             throw handleException(e, runtime, strategy);
         }
     }
+    
+	@Override
+	public IStrategoTerm invoke(HybridInterpreter runtime, IStrategoTerm input, String strategy,
+			List<IStrategoTerm> termArguments) throws MetaborgException {
+		runtime.setCurrent(input);
+		
+		final IStrategoString strategyName = createStrategyName(strategy, termArguments);
+		final IStrategoAppl strategyNameTerm = termFactory.makeAppl("SVar", strategyName);
+		final IStrategoAppl strategyCallTerm = termFactory.makeAppl("CallT", strategyNameTerm, termFactory.makeList(), termFactory.makeList(termArguments));
+		
+        final IStrategoConstructor sdefT = termFactory.makeConstructor("SDefT", 4);
+		final IStrategoAppl strategyTerm = termFactory.makeAppl(sdefT, strategyName, 
+				termFactory.makeList(), termFactory.makeList(), strategyCallTerm);
+		try {
+			if(runtime.evaluate(strategyTerm)) {
+				return runtime.current();
+			}
+		} catch (InterpreterException e) {
+            throw handleException(e, runtime, strategy);
+		}
+		return null;
+	}
+		
+	private IStrategoString createStrategyName(String identifier, List<IStrategoTerm> termArguments) {
```

Comment:
It would make more sense to me if the size is passed as an argument instead of the whole list.

Reply:
That would just move the null check at the call site of the function. I thought it would be better to also encapsulate that in the helper function to keep the invoke method cleaner.

Reply:
I don't think you need a null check. The convention in our code is to tag types with `@Nullable` and assume non-null otherwise. It makes sense to me to require the caller not to call the method with a null for the list.

Reply:
You're right, it doesn't really make sense to call this function with null for term arguments anyway, since there are other invoke methods to call strategies without term arguments. I removed the null check as you suggested.